### PR TITLE
support x-ms-client-default for property

### PR DESCRIPTION
**Pull request by haolingdong-msft**
Code change:
1. add clientDefaultValue in  codemodel/Property.java
2. handle x-ms-client-default in ModelPropertyMapper.java
3. handle x-ms-client-default in ModelTemplate.java
4. add vanilla test
5. add fluent  test

Generated class:
vanilla-tests/src/main/java/fixtures/clientdefaultvalue/models/Sku.java

### CODE REVIEW

File: extension-base/src/main/java/com/azure/autorest/extension/base/model/codemodel/Property.java [hunk 1]
```diff
@@ -128,4 +130,20 @@ public void setParentSchema(ObjectSchema parentSchema) {
...
+
+    public String getClientDefaultValue() {
+        if (clientDefaultValue == null) {
+            if (this.getSchema() != null && this.getSchema() instanceof ConstantSchema) {
+                ConstantSchema constantSchema = (ConstantSchema) this.getSchema();
+                if (constantSchema.getValue() != null) {
+                    this.setClientDefaultValue(constantSchema.getValue().getValue().toString());
+                }
+            }
+        }
```
[the remark this answers is not in the document]
Comment (haolingdong-msft) [reply extension-base/src/main/java/com/azure/autorest/extension/base/model/codemodel/Property.java:142]: For this part, I referred to Parameter.java: https://github.com/Azure/autorest.java/blob/main/extension-base/src/main/java/com/azure/autorest/extension/base/model/codemodel/Parameter.java#L21

File: vanilla-tests/swagger/client-default-value.json [hunk 2]
```diff
@@ -0,0 +1,76 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2021-09-15",
+    "title": "MonitorManagementClient",
```
[the remark this answers is not in the document]
Comment (haolingdong-msft) [reply vanilla-tests/swagger/client-default-value.json:5]: will update, thx

File: extension-base/src/main/java/com/azure/autorest/extension/base/model/codemodel/Property.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (weidongxu-microsoft) [reply extension-base/src/main/java/com/azure/autorest/extension/base/model/codemodel/Property.java:142]: Same comment, this is not a good approach. Prefer do this in Mapper.

[the remark this answers is not in the document]
Comment (haolingdong-msft) [reply extension-base/src/main/java/com/azure/autorest/extension/base/model/codemodel/Property.java:142]: ok, will update

File: javagen/src/main/java/com/azure/autorest/mapper/ModelPropertyMapper.java [hunk 3]
```diff
@@ -170,6 +170,14 @@ public ClientModelProperty map(Property property) {
         }
 
         // handle x-ms-client-default for primitive type, enum, boxed type and string
+        if (property.getClientDefaultValue() == null) {
+            if (property.getSchema() != null && property.getSchema() instanceof ConstantSchema) {
+                ConstantSchema constantSchema = (ConstantSchema) property.getSchema();
+                if (constantSchema.getValue() != null) {
+                    property.setClientDefaultValue(constantSchema.getValue().getValue().toString());
```
Comment (weidongxu-microsoft) [inline javagen/src/main/java/com/azure/autorest/mapper/ModelPropertyMapper.java:177]: Don't set it back, just use your result. `Property` is data model, consider it as immutable.

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**